src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = hash(key);
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
 
+        if (table[hash] == null) {
+            table[hash] = newNode;
+        } else {
+            Node<K, V> current = table[hash];
+            while (current != null) {
+                if (current.hash == hash
+                        && (current.key == key || (key != null && key.equals(current.key)))) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = newNode;
+                    break;
+                }
+                current = current.next;
+            }
+        }
+
+        size++;
+        resizeIfNeeded();
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        Node<K, V> node = table[hash];
+
+        while (node != null) {
+            if (node.hash == hash
+                    && (node.key == key || (key != null && key.equals(node.key)))) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % DEFAULT_CAPACITY);
+    }
+
+    private void resizeIfNeeded() {
+        if (size > threshold) {
+            resize();
+        }
+    }
+
+    private void resize() {
+        int newCapacity = table.length * 2;
+        threshold = (int) (newCapacity * LOAD_FACTOR);
+        Node<K, V>[] newTable = new Node[newCapacity];
+
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                int newHash = Math.abs(node.hash % newCapacity);
+                Node<K, V> next = node.next;
+                node.next = newTable[newHash];
+                newTable[newHash] = node;
+                node = next;
+            }
+        }
+        table = newTable;
+    }
+
+    static class Node<K, V> {
+        private final int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
```

Comment:
Code style issue: According to the checklist, the inner class Node should be parameterized and should not use getters and setters. However, in your implementation, the Node class is already parameterized and does not use getters and setters, which is correct. But the fields `key`, `value` and `next` should be final to ensure immutability. Only the `value` field can be non-final because it needs to be updated when a new value is put for the same key.